lib/Sema/TypeCheckType.cpp

```diff
@@ -4776,7 +4777,9 @@ NeverNullType TypeResolver::resolveVarargType(VarargTypeRepr *repr,
   // do not allow move-only types as the element of a vararg
   if (!element->hasError()
       && inStage(TypeResolutionStage::Interface)
-      && element->isNoncopyable(getDeclContext())) {
+      && !options.contains(TypeResolutionFlags::SILMode)
+      && isInterfaceTypeNoncopyable(
+          element, getDeclContext()->getGenericEnvironmentOfContext())) {
```

Comment:
This should also be changed to options.getGenericSignature().getGenericEnvironment()